src/pages/dashboard.tsx

```diff
@@ -17,20 +20,86 @@ import {
 import Link from "next/link";
 import { DateSelect } from "../components/DateSelect";
 import { currentTimezone } from "../utils/timezone";
+import { id } from "date-fns/locale";
+interface Event {
+  id: string;
+  name?: string;
+  occuringDays?: string;
+  joinCode: string;
+  begins?: Date;
+  ends?: Date;
+  // Add other properties as needed
+}
+interface EventCardProps {
+  event: Event;
+}
+
+const EventCard: React.FC<EventCardProps> = ({ event }) => {
+  console.log("Event data:", event);
+  const eventId = event.id;
+  const eventName = event.name;
+  const occuringDaysArray = event.occuringDays?.split(",");
+  const startTime = event.begins;
+  const endTime = event.ends;
+
+  // Convert the Date objects to the desired format
+  const formatTime = (date: Date) => {
+    try {
+      return new Intl.DateTimeFormat("en-US", {
+        hour: "numeric",
+        minute: "2-digit",
+        hour12: true,
+      }).format(date);
+    } catch (error) {
+      console.error("Invalid time value:", date, error);
+      return "Invalid time";
+    }
+  };
+
+  const formatDate = (date: Date) => {
+    try {
+      return new Intl.DateTimeFormat("en-US", {
+        weekday: "short",
+        month: "short",
+        day: "2-digit",
+        year: "numeric",
+      }).format(date);
+    } catch (error) {
+      console.error("Invalid date value:", date, error);
+      return "Invalid date";
+    }
+  };
```

Comment:
The function is duplicated here. Also, using the `format` function from the `date-fns` package is preferable. See documentation: https://date-fns.org/v2.30.0/docs/format.

Reply:
I will be working on this.